File: app/src/main/java/com/fsoon/android/fsfirsttemplate/view/tamplate/ScrollingHeaderActivity.java
package com.fsoon.android.fsfirsttemplate.view.tamplate;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.fsoon.android.fsfirsttemplate.R;
import com.fsoon.android.fsfirsttemplate.view.base.BaseActivity;

public class ScrollingHeaderActivity extends BaseActivity implements View.OnClickListener {

    private ImageView mLeftButton, mRightButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_scrolling_header);
    }

    public void init(String title) {
        initButton();
        setTitle(title);
    }

    public void init(String title, boolean lVisible, boolean rVisible) {
        initButton();
        setTitle(title);
        setLeftButtonVisible(lVisible);
        setRightButtonVisible(rVisible);
    }

    public void init(String title, boolean lVisible, boolean rVisible, boolean lEnable, boolean rEnable) {
        initButton();
        setTitle(title);
        setLeftButtonVisible(lVisible);
        setRightButtonVisible(rVisible);
        setLeftButtonEnable(lEnable);
        setRightButtonEnable(rEnable);
    }

    public void setTitle(String title) {
        TextView titleTextView = findViewById(R.id.titleTextView);
        titleTextView.setText(title);
    }

    private void initButton() {
        mLeftButton = findViewById(R.id.rightButton);
        mRightButton = findViewById(R.id.leftButton);
        mLeftButton.setOnClickListener(this);
        mRightButton.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.leftButton:
                break;
            case R.id.rightButton:
                break;
        }
    }

    public void setLeftButtonVisible(boolean isVisible) {
        mLeftButton.setVisibility(isVisible ? View.VISIBLE : View.INVISIBLE);
        mLeftButton.setClickable(isVisible);
    }

    public void setRightButtonVisible(boolean isVisible) {
        mRightButton.setVisibility(isVisible ? View.VISIBLE : View.INVISIBLE);
        mRightButton.setClickable(isVisible);
    }

    public void setLeftButtonEnable(boolean isEnable) {
        mLeftButton.setEnabled(isEnable);
    }

    public void setRightButtonEnable(boolean isEnable) {
        mRightButton.setEnabled(isEnable);
    }
}

File: app/src/main/kotlin/com/fsoon/android/fsfirsttemplate/view/tamplate/ScrollingHeaderActivity.kt
package com.fsoon.android.fsfirsttemplate.view.tamplate

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.fsoon.android.fsfirsttemplate.R
import com.fsoon.android.fsfirsttemplate.view.base.BaseActivity

class ScrollingHeaderActivity : BaseActivity(), View.OnClickListener {
    private lateinit var mLeftButton: ImageView
    private lateinit var mRightButton: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_scrolling_header)
    }

    fun init(title: String?) {
        initButton()
        setTitle(title)
    }

    fun init(title: String?, lVisible: Boolean, rVisible: Boolean) {
        initButton()
        setTitle(title)
        setLeftButtonVisible(lVisible)
        setRightButtonVisible(rVisible)
    }

    fun init(
        title: String?,
        lVisible: Boolean,
        rVisible: Boolean,
        lEnable: Boolean,
        rEnable: Boolean
    ) {
        initButton()
        setTitle(title)
        setLeftButtonVisible(lVisible)
        setRightButtonVisible(rVisible)
        setLeftButtonEnable(lEnable)
        setRightButtonEnable(rEnable)
    }

    fun setTitle(title: String?) {
        val titleTextView = findViewById<TextView>(R.id.titleTextView)
        titleTextView.text = title
    }

    private fun initButton() {
        mLeftButton = findViewById(R.id.rightButton)
        mRightButton = findViewById(R.id.leftButton)
        mLeftButton.setOnClickListener(this)
        mRightButton.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.leftButton -> {}
            R.id.rightButton -> {}
        }
    }

    fun setLeftButtonVisible(isVisible: Boolean) {
        mLeftButton.visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
        mLeftButton.isClickable = isVisible
    }

    fun setRightButtonVisible(isVisible: Boolean) {
        mRightButton.visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
        mRightButton.isClickable = isVisible
    }

    fun setLeftButtonEnable(isEnable: Boolean) {
        mLeftButton.isEnabled = isEnable
    }

    fun setRightButtonEnable(isEnable: Boolean) {
        mRightButton.isEnabled = isEnable
    }
}